src/darwin/Framework/CHIP/templates/partials/attribute_data_callback_name.zapt

```diff
@@ -7,13 +7,21 @@
          than a per-struct-type basis. ~}}
     {{asUpperCamelCase parent.name}}{{asUpperCamelCase name}}Struct
   {{~else if_is_bitmap type~}}
-    {{asUpperCamelCase parent.name}}{{asUpperCamelCase name}}
+    {{~#if_is_atomic type~}}
```

Comment:
Again, checking atomic here is not what we want.  Same in other places.

Reply:
Before atomic enums were not part of the enums table. With the new schema all enums atomic or non atomic both are part of the enum table. Do you want me to create a single helper which combines both?

Reply:
My point is that "atomic" or not is not a useful distinction for these templates.  What they want to know is whether the C++ type is a raw int or a strongly typed enum/bitmap.  We should have a helper that answers that question.  It would return true for things that are non-atomic or vendor_id (for enums), false for others.

Reply:
I updated these with `if_is_non_atomic_enum` and `if_is_non_atomic_bitmap`. I still feel vendor id should not be hardcoded since it is confusing(I moved that logic into the templates. Not that, that is great either). I feel we should have the c++ files use a different zcl.json/types.xml with vendor Id being an enum. This way we do not have hardcoded data type handling which leads to users having to debug why this one type behaves different.

Reply:
No one reading these templates will have any idea what "atomic" means.  What they do know, and what the templates care about, is what the C++ representation is.

Reply:
Cleaned it up. Let me know if the current changes seem fine.